flaml/automl/time_series/ts_model.py

```diff
@@ -0,0 +1,760 @@
+            train_df = data.train_data[regressors + [target_col, time_col]]
+            train_df = train_df.rename(columns={target_col: "y", time_col: "ds"})
+        else:
+            train_df = self._join(X_train, y_train)
+
+            regressors = list(train_df.columns)
+            regressors.remove(TS_TIMESTAMP_COL)
+            regressors.remove(TS_VALUE_COL)
+
+        train_df = self._preprocess(train_df)
+        logging.getLogger("prophet").setLevel(logging.WARNING)
+        nice_params = {k: v for k, v in self.params.items() if k in self._search_space()}
+        model = Prophet(**nice_params)
+        for regressor in regressors:
+            model.add_regressor(regressor)
+        with suppress_stdout_stderr():
+            model.fit(train_df)
+        train_time = time.time() - current_time
+        self._model = model
+        return train_time
+
+    def predict(self, X, **kwargs):
+        X = self.enrich(X)
+        if isinstance(X, int):
+            raise ValueError(
+                "predict() with steps is only supported for arima/sarimax."
+                " For Prophet, pass a dataframe with the first column containing"
+                " the timestamp values."
+            )
+
+        if isinstance(X, TimeSeriesDataset):
+            data = X
+            X = data.test_data[data.regressors + [data.time_col]]
+
+        X = X.rename(columns={self.time_col: "ds"})
+        if self._model is not None:
+            X = self._preprocess(X)
+            forecast = self._model.predict(X, **kwargs)
+            out = forecast["yhat"]
+            out.name = self.target_names[0]
+            return out
+
+        else:
+            logger.warning("Estimator is not fit yet. Please run fit() before predict().")
+            return np.ones(X.shape[0])
+
+
+class StatsModelsEstimator(TimeSeriesEstimator):
+    def predict(self, X, **kwargs) -> pd.Series:
+        X = self.enrich(X)
+        if self._model is None or self._model is False:
+            return np.ones(X if isinstance(X, int) else X.shape[0])
+
+        if isinstance(X, int):
+            return self._model.forecast(steps=X)
+
+        if isinstance(X, TimeSeriesDataset):
+            data = X
+            X = data.test_data[data.regressors + [data.time_col]]
+        else:
+            X = X[self.regressors + [self.time_col]]
+
+        if isinstance(X, DataFrame):
+            start = X[self.time_col].iloc[0]
+            end = X[self.time_col].iloc[-1]
+            if len(self.regressors):
+                exog = self._preprocess(X[self.regressors])
+                forecast = self._model.predict(start=start, end=end, exog=exog.values, **kwargs)
+            else:
+                forecast = self._model.predict(start=start, end=end, **kwargs)
+        else:
+            raise ValueError(
+                "X needs to be either a pandas Dataframe with dates as the first column"
+                " or an int number of periods for predict()."
+            )
```

Comment:
This is an example of code path covered by the commented test code.

Reply:
OK. put the test back in.